roles/openshift_storage_glusterfs/tasks/heketi_deploy_part1.yml

```diff
@@ -0,0 +1,41 @@
+---
+- name: Copy initial heketi resource files
+  copy:
+    src: "{{ openshift.common.examples_content_version }}/{{ item }}"
+    dest: "{{ mktemp.stdout }}/{{ item }}"
+  with_items:
+  - "deploy-heketi-template.yml"
+
+- name: Create deploy-heketi resources
+  oc_obj:
+    namespace: "{{ openshift_storage_glusterfs_namespace }}"
+    kind: template
+    name: deploy-heketi
+    state: present
+    files:
+    - "{{ mktemp.stdout }}/deploy-heketi-template.yml"
+
+- name: Deploy deploy-heketi pod
+  oc_process:
+    namespace: "{{ openshift_storage_glusterfs_namespace }}"
+    template_name: "deploy-heketi"
+    create: True
+    params:
+      IMAGE_NAME: "{{ openshift_storage_glusterfs_heketi_image }}"
+      IMAGE_VERSION: "{{ openshift_storage_glusterfs_heketi_version }}"
+      HEKETI_USER_KEY: "{{ openshift_storage_glusterfs_heketi_user_key }}"
+      HEKETI_ADMIN_KEY: "{{ openshift_storage_glusterfs_heketi_admin_key }}"
+
+- name: Wait for deploy-heketi pod
+  oc_obj:
+    namespace: "{{ openshift_storage_glusterfs_namespace }}"
+    kind: pod
+    state: list
+    selector: "glusterfs=deploy-heketi-pod,deploy-heketi=support"
+  register: heketi_pod
+  until:
```

Comment:
Agreed. Didn't want to try and do that here given the timeline I'm on, but that's what I'd rather do. The tricky part is that a watch/wait operation might have to be different for every kind of resource.